lib/selects.bzl

```diff
@@ -76,7 +76,171 @@ def _with_or_dict(input_dict):
+        native.alias(
+            name = name,
+            actual = settings[0],
+        )
+        return
+
+    # We need n-1 aliases for n settings. The first alias has no extension. The
+    # second alias is named name + "_2", and so on. For the first n-2 aliases,
+    # if they don't match they reference the next alias over. If the n-1st alias
+    # doesn't match, it references the final setting (which is then evaluated
+    # directly to determine the final value of the AND chain).
+    actual = [name + "_" + str(i) for i in range(2, len(settings))]
+    actual.append(settings[-1])
+
+    for i in range(1, len(settings)):
+        native.alias(
+            name = name if i == 1 else name + "_" + str(i),
+            actual = select({
+                settings[i - 1]: settings[i - 1],
+                "//conditions:default": actual[i - 1],
+            }),
+        )
+
+def _config_setting_and_group(name, settings):
+    """ANDs multiple config_settings together.
+
+    The core idea is to create a sequential chain of alias targets where each is
+    select-resolved as follows: If alias n matches config_setting n, it resolves to
+    alias n+1 (which evaluates config_setting n+1, and so on). Else it resolves to
+    config_setting n, which doesn't match by definition. The only way to get a
+    matching final result is if all config_settings match.
+    """
+
+    # "//conditions:default" is automatically true so doesn't need checking.
+    settings = _remove_default_condition(settings)
+
+    # One config_setting input? Just alias directly to it.
+    if len(settings) == 1:
+        native.alias(
+            name = name,
+            actual = settings[0],
+        )
+        return
+
+    # We need n-1 aliases for n settings. The first alias has no extension. The
+    # second alias is named name + "_2", and so on. For the first n-2 aliases,
+    # if they match they reference the next alias over. If the n-1st alias matches,
+    # it references the final setting (which is then evaluated directly to determine
+    # the final value of the AND chain).
+    actual = [name + "_" + str(i) for i in range(2, len(settings))]
+    actual.append(settings[-1])
+
+    for i in range(1, len(settings)):
+        native.alias(
+            name = name if i == 1 else name + "_" + str(i),
+            actual = select({
+                settings[i - 1]: actual[i - 1],
+                "//conditions:default": settings[i - 1],
+            }),
+        )
+
+def _config_setting_always_true(name):
+    """Returns a config_setting with the given name that's always true.
+
+    This is achieved by constructing a two-entry OR chain where each
+    config_setting takes opposite values of a boolean flag.
+    """
+    name_on = name + "_stamp_binary_on_check"
+    name_off = name + "_stamp_binary_off_check"
+    native.config_setting(
+        name = name_on,
+        values = {"stamp": True},
```

Comment:
When I try to manually create such a rule, Bazel complains
```
expected value of type 'string' for dict value element, but got True (bool)
```
so it seems like this is not valid. Is it possible this function is not being exercised?

Reply:
You're absolutely right. That's a bug and I didn't add coverage for `//conditions:default`.